fhttp/http_client.go

```diff
@@ -274,10 +274,19 @@ func (h *HTTPOptions) AddAndValidateExtraHeader(hdr string) error {
 	}
 	key := strings.TrimSpace(s[0])
 	value := strings.TrimSpace(s[1])
-	if strings.EqualFold(key, "host") {
+	switch strings.ToLower(key) {
+	case "host":
 		log.LogVf("Will be setting special Host header to %s", value)
 		h.hostOverride = value
-	} else {
+	case "user-agent":
+		if value == "" {
```

Comment:
I think we should delete this part and allow setting an empty user-agent. Apparently this is some use-case where people _dont_ want to set a header but requests get proxied. That proxy adds a user agent if it's not set. Therefore apparently there is a use-case for people to set an empty user-agent header. 

https://github.com/zalando/skipper/issues/1994

Reply:
So my goal here is to provide a way for people to use `fortio curl` and delete the (otherwise present by default) `User-Agent:` header (without adding yet another flag if at all possible as we do have too many already)

My first change only had the Set part but then I noticed on the wire sending an empty
```
User-Agent:
```
which waste bytes and looks ugly - also on the server side can't be (in go at least) distinguished from absent

I see that use case to be a lot more practical/useful than if a proxy has some behavior that can be argued as bugged (why add a UA header instead of a proxy-version one?)

Reply:
the go bug is interesting though (and that's what it is imo, a go bug, to decide to add UA when missing in a reverse proxy) - thanks for pointing that out as it applies to `fortio server -M` as well as https://github.com/fortio/proxy

Reply:
made some adjustment (yet also didn't yet offer the option to send an empty User-Agent)

Reply:
Ok with latest update you can use white space to send an empty User-Agent: and empty for delete/no such header. (and avoids getting `User-Agent: Go-http-client/1.1` either)

Reply:
@wiardvanrij ready for final review now - I think - tia

Reply:
> funny how something small ends up into this :)

indeed it was quite a long winded road - thanks for pointing the issues out